datafeeder/src/main/java/org/georchestra/datafeeder/service/publish/mock/MockDataBackendService.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2020, 2021 by the geOrchestra PSC
+ *
+ * This file is part of geOrchestra.
+ *
+ * geOrchestra is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * geOrchestra is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * geOrchestra. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.georchestra.datafeeder.service.publish.mock;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.georchestra.datafeeder.model.DataUploadJob;
+import org.georchestra.datafeeder.model.DatasetUploadState;
+import org.georchestra.datafeeder.model.UserInfo;
+import org.georchestra.datafeeder.service.DatasetsService;
+import org.georchestra.datafeeder.service.publish.DataBackendService;
+import org.geotools.data.shapefile.ShapefileDirectoryFactory;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.FileSystemUtils;
+
+import lombok.NonNull;
+
+public class MockDataBackendService implements DataBackendService, DisposableBean {
```

Comment:
you do need the Mock* classes into the src/ subdir ? (not into test/ ?)

Reply:
Actually we used to, because the different parts of the system were mocked up until we added them one by one (e.g. database, geoserver, geonetwork interactors). Good point though, now they should be in `src/test`

Reply:
done. b28c9a13780269e5a1572324f79bef57e6528c6a